packages/http-client/src/http-client.spec.ts

```diff
@@ -305,6 +319,77 @@ describe('generic http client', () => {
       expect(response.data).toEqual('Initial value.Middleware One.');
     });
 
+    function mockJwtValidationAndServiceToken() {
+      const jku = 'https://my-jku-url.authentication.sap.hana.ondemand.com';
+      nock(jku).get('/').reply(200, responseWithPublicKey());
+      mockUserTokenGrantCall(
+        providerXsuaaUrl,
+        1,
+        subscriberUserJwt,
+        subscriberUserJwt,
+        xsuaaBindingMock.credentials
+      );
+      mockClientCredentialsGrantCall(
+        providerXsuaaUrl,
+        { access_token: providerServiceToken },
+        200,
+        destinationBindingClientSecretMock.credentials
+      );
+      mockClientCredentialsGrantCall(
+        subscriberXsuaaUrl,
+        { access_token: subscriberServiceToken },
+        200,
+        destinationBindingClientSecretMock.credentials
+      );
+    }
+
+    function mockDestinationService() {
+      mockInstanceDestinationsCall(nock, [], 200, subscriberServiceToken);
+      mockSubaccountDestinationsCall(
+        nock,
+        basicMultipleResponse,
+        200,
+        subscriberServiceToken
+      );
+    }
+
+    it('passes the context properties to the middleware', async () => {
+      const showContextMiddleware: Middleware<
+        HttpResponse,
+        HttpMiddlewareContext
+      > = (opt: MiddlewareIn<HttpResponse, HttpMiddlewareContext>) => () =>
+        ({ data: opt.context } as any);
+
+      mockServiceBindings();
+      // Inside connectivity package we can use jest to mock jwtVerify and tokenAccessor
+      // Between modules this is not possible and done via HTTP mocks instead
+      mockJwtValidationAndServiceToken();
+      mockDestinationService();
+
+      const response = await executeHttpRequest(
+        {
+          destinationName: 'FINAL-DESTINATION',
+          jwt: subscriberUserJwt,
+          iasToXsuaaTokenExchange: false
+        },
+        {
+          middleware: [showContextMiddleware],
+          method: 'get'
+        }
+      );
+      expect(response.data).toEqual({
+        destinationName: 'FINAL-DESTINATION',
+        jwt: subscriberUserJwt,
+        requestConfig: expect.any(Object),
+        tenantId: 'subscriber',
+        uri: 'https://my.system.com'
```

Comment:
```suggestion
        uri: 'https://my.system.example.com'
```
[req] Always use example.com or an SAP URL